webapp-mgmt/cas-management-webapp-configuration/src/main/java/org/apereo/cas/mgmt/configuration/CasConfigurationPropertiesEnvironmentManager.java

```diff
@@ -0,0 +1,108 @@
+package org.apereo.cas.mgmt.configuration;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.FileBasedConfiguration;
+import org.apache.commons.configuration2.PropertiesConfiguration;
+import org.apache.commons.configuration2.builder.FileBasedConfigurationBuilder;
+import org.apache.commons.configuration2.builder.fluent.Parameters;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apereo.cas.configuration.CasConfigurationProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.env.Environment;
+import org.springframework.util.Assert;
+
+import java.io.File;
+import java.util.Map;
+
+/**
+ * This is {@link CasConfigurationPropertiesEnvironmentManager}.
+ *
+ * @author Misagh Moayyed
+ * @since 5.1.0
+ */
+public class CasConfigurationPropertiesEnvironmentManager {
```

Comment:
This file need to go away. You should use one that is provided to you by the configuration-api module.

Reply:
What about the rebind method that uses CasConfigurationProperties?

Reply:
That I don't think you need either...or at least not right away. rebind is mostly used to pick up changes done to the config file and dynamically rebind objects to values and refresh the application context. I'd say that's not a very immediate use case for the management app.